### PR TITLE
Integrate docker

### PR DESCRIPTION
**Pull request by vladgabriel**
Hi there,

I've created a Dockerfile together with a docker-compose example usage. 
Some details:
- There is no spamassassin integration
- It is mandatory to setup two mx records in environment variables, otherwise the app will crash with:  _550 this.config.exchanges.every is not a function_ 
- Spf ain't working due to docker network, haven't invested time to see a viable solution
- It is assumed you have a dkim folder with the private key, check docker-compose
- It is assumed you have certbot on host, check docker-compose
- Proxy server should be ignore
- Redis is in another container, SMTP_REDIS_HOST is needed, might be also necessary to add it to .env.schema

Any suggestion is welcome.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+FROM node:lts-alpine
+
+RUN \
+  apk update && \
```
Comment (aslafy-z) [inline Dockerfile]: This is not needed when using `--no-cache` since package list is downloaded every time.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  apk add --no-cache \
+    python3 \
+    && \
+  cp /usr/bin/python3 /usr/bin/python && \
```
Comment (aslafy-z) [inline Dockerfile]: Prefer symlinking over copying. (`ln -s`)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  pip3 install \
```
Comment (aslafy-z) [inline Dockerfile]: You can add `--no-cache-dir` to prevent pip from keeping cache files.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    pyspf \
+    dnspython \
+    dkimpy
+
+WORKDIR /app
+
+COPY package*.json ./
```
Comment (aslafy-z) [inline Dockerfile]: Can be factored to `COPY package.json yarn.lock ./`

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+COPY yarn.lock yarn.lock
+
+RUN npm i -g pm2 && npx yarn --prod
```
Comment (aslafy-z) [inline Dockerfile]: - install global packages before copying over package.json (move this statement after pip install)
- yarn is bundled in this image, no need to use npx.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+COPY index.js index.js
```
Comment (aslafy-z) [inline Dockerfile]: I'd go for the inverse strategy, ignore files that aren't needed in the image into the `.dockerignore` and then do a ` COPY . ./`

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+COPY app.js app.js
+COPY helpers/* ./helpers/
+COPY .env.* ./
+COPY ecosystem.json ecosystem.json
+
+ENV NODE_ENV production
```
Comment (aslafy-z) [inline Dockerfile]: This can go top of the file

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+EXPOSE 25
```
Comment (aslafy-z) [inline Dockerfile]: These can be defined in the same EXPOSE directive.

**Comment by aslafy-z** [comment]
One more thing, I think pm2 is not needed. The app will run in a container which have to crash when the soft inside does so.

**Comment by vladgabriel** [comment]
Follow-up in #210.

**Pull request closed without merging**